### PR TITLE
refactoring for easy unit-test and cm-cli integration

### PR DESCRIPTION
**Pull request by itdove**
Refactoring for easy unit-tests and cm-cli integration
Also quick fix for : https://github.com/open-cluster-management-io/clusteradm/issues/12
Remove the "verbs" notion, use groups like oc
Signed-off-by: Dominique Vernier <dvernier@redhat.com>

### CODE REVIEW

File: pkg/cmd/join/exec.go [hunk 1]
```diff
@@ -90,18 +92,22 @@ func (o *Options) run() error {
...
 	}
 
-	err = helpers.ApplyDeployment(kubeClient, reader, scenarioDirectory, o.values, "join/operator.yaml")
+	err = apply.ApplyDeployment(kubeClient, reader, o.values, "", "join/operator.yaml")
 	if err != nil {
 		return err
 	}
 
+	//quick fix for https://github.com/open-cluster-management-io/clusteradm/issues/12
+	fmt.Printf("Wait 10 sec... for the crd to be effective\n")
```
Comment (qiujian16) [inline pkg/cmd/join/exec.go:106]: you could use wait.Until here to get crd  of clustermanager.

File: pkg/helpers/apply/apply.go [hunk 2]
```diff
@@ -159,8 +161,18 @@ func getTemplate(templateName string) *template.Template {
 	return tmpl
 }
 
-func MustTempalteAsset(name, templateName string, reader ScenarioReader, values interface{}) ([]byte, error) {
-	tmpl := getTemplate(templateName)
+//MustTempalteAsset generates textual output for an file name.
+//If a headerFile is specified it will be added as a header of the file.
```
Comment (qiujian16) [inline pkg/helpers/apply/apply.go:165]: I am not quite sure what is the purpose of headerfile. Would you add more comments here?

Comment (itdove) [reply pkg/helpers/apply/apply.go:165]: The header file is an extra file you would like to add to the the files... this file can contain template functions... this like having the helm _helpers.tpl.

File: pkg/cmd/join/exec.go (hunk 1 above)
Comment (itdove) [reply pkg/cmd/join/exec.go]: implement retry.OnError

**Review by qiujian16 — COMMENTED** [review]
/lgtm

**Pull request merged**